docs/asset/operations/sample_exact_es_percent.md

```diff
@@ -0,0 +1,110 @@
+# sample_exact_es_percent
+
+Given an array of JSON documents will return an array containing a shuffled subset of those input documents. The size of the subset will be the percentage stored in an elasticsearch record multiplied against the length of the array rounded down. Percentage will be recalculated every `window_ms` milliseconds by querying the elasticsearch located at `connection` for the document with `document_id` found within the `index` specified in the config.
+
+## Usage
+
+In this example we want to retrieve the percentage from the document with _id `abc123` in index `my-index` at the terafoundation connection named `elasticsearch7`.
+
+### Example of a job using the `sample_exact_es_percent` processor
+
+```json
+{
+    "name" : "testing",
+    "workers" : 1,
+    "slicers" : 1,
+    "lifecycle" : "once",
+    "assets" : [
+        "standard"
+    ],
+    "operations" : [
+        {
+            "_op": "test-reader"
+        },
+        {
+            "_op": "sample_exact_es_percent",
+            "connection": "elasticsearch7",
+            "index": "my-index",
+            "document_id": "abc123",
+            "window_ms": 300000
+        }
+    ]
+}
+
+```
+
+### Example sample index creation
+
+Create an index with 1 shard, 2 replicas, and a single text property called `percent`.
+
+```sh
+curl -X PUT "http://localhost:9200/my-index" -H 'Content-Type: application/json' -d'
+{
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 2
+  },
+  "mappings": {
+    "properties": {
+      "percent": {
+        "type": "text"
+      }
+    }
+  }
+}'
+```
+
+### Example document creation
+
+```sh
+curl -X POST 'localhost:9200/my-index/_doc/abc123' -H 'Content-Type:application/json' -d '
+{
+  "percent": 50
```

Comment:
If we ever wanted to re-index this index (move it to another cluster perhaps) it would not be possible to use Teraslice because the ES reader in Teraslice requires either `_key` (or some ID field) or timestamps on the records.  Let's ignore that for now.

Reply:
I think we could update these docs to show what the desired document should look like. The processor only needs the `percent` field, but the controller I am building adds a timestamp and a reference to the index it is tracking. I could also add a `_key` field.